### PR TITLE
User-Index/Show

### PR DESCRIPTION
**Pull request by TrucLuu**
Update show-index user

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -7,6 +7,13 @@ def new
     @user = User.new
   end
 
+  def index
```
Comment (phamkykhoi) [inline app/controllers/users_controller.rb]: Move action index to before action new

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -7,6 +7,13 @@ def new
...
+    @users = User.all
+  end
+
+  def show
+  end
+
   def create
     @user = User.new user_params
     if @user.save
```
Comment (phamkykhoi) [inline app/controllers/users_controller.rb]: Remove ( ) after current_user?

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -1,12 +1,19 @@
 class UsersController < ApplicationController
-  before_action :logged_in_user, only: [:edit, :update]
-  before_action :correct_user, only: [:edit, :update]
-  before_action :set_user, only: [:show, :edit, :update]
+  before_action :logged_in_user, only : [:edit, :update]
+  before_action :correct_user, only : [:edit, :update]
+  before_action :set_user, only : [:show, :edit, :update]
+
+  def index
+    @users = User.all
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: pagination

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -1,12 +1,19 @@
 class UsersController < ApplicationController
-  before_action :logged_in_user, only: [:edit, :update]
-  before_action :correct_user, only: [:edit, :update]
-  before_action :set_user, only: [:show, :edit, :update]
+  before_action :logged_in_user, only : [:edit, :update]
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: `only:` not `only :`

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -1,12 +1,19 @@
...
+  before_action :correct_user, only : [:edit, :update]
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: `only:` not `only :`

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -1,12 +1,19 @@
...
+  before_action :set_user, only : [:show, :edit, :update]
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: `only:` not `only :`

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -3,10 +3,17 @@ class UsersController < ApplicationController
   before_action :correct_user, only: [:edit, :update]
   before_action :set_user, only: [:show, :edit, :update]
 
+  def index
+    @users = User.paginate(page: params[:page], per_page: 10)
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: remove `()`

File: app/controllers/users_controller.rb [hunk 8]
```diff
@@ -3,10 +3,17 @@ class UsersController < ApplicationController
   before_action :correct_user, only: [:edit, :update]
   before_action :set_user, only: [:show, :edit, :update]
 
+  def index
+    @users = User.paginate page: params[:page], per_page: 10
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: Store `10 `in Settings

**Comment by dieunb** [comment]
OK

**Pull request merged**